### PR TITLE
Issue 6119: Large Append Wirecommand

### PR DESCRIPTION
**Pull request by co-jo**
**Change log description**  

* Adds new WireCommand(s) and create method stubs in appropriate classes.

**Purpose of the change**  

Finishes #6116

### CODE REVIEW

**Review by andreipaduroiu — CHANGES_REQUESTED** [review]
This PR seems like it has a lot of unrelated changes into it.

Can you please update the base branch with the latest master and then rebase/remerge your PR so we don't see all this health check changes too?

**Comment by co-jo** [comment]
> 
> 
> This PR seems like it has a lot of unrelated changes into it.
> 
> Can you please update the base branch with the latest master and then rebase/remerge your PR so we don't see all this health check changes too?

Yes, my mistake.

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java [hunk 1]
```diff
@@ -1171,6 +1171,62 @@ public static WireCommand readFrom(DataInput in, int length) throws IOException
         }
     }
 
+    @Data
+    public static final class CreateTransientSegment implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.CREATE_TRANSIENT_SEGMENT;
+        final long requestId;
+        final String segmentName;
```
Comment (tkaitchuck) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:1179]: parentSegment?

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java [hunk 2]
```diff
@@ -1171,6 +1171,62 @@ public static WireCommand readFrom(DataInput in, int length) throws IOException
         }
     }
 
+    @Data
+    public static final class CreateTransientSegment implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.CREATE_TRANSIENT_SEGMENT;
+        final long requestId;
```
Comment (tkaitchuck) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:1178]: Should this contain a writerId?

File: shared/protocol/src/test/java/io/pravega/shared/protocol/netty/DelegatingRequestProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import org.junit.Test;
+
+public class DelegatingRequestProcessorTest {
+
+    private static final Event EMPTY_EVENT = new WireCommands.Event(Unpooled.wrappedBuffer(new byte[1]));
+
+    private MockRequestProcessor rp = new MockRequestProcessor();
+
+    @Test
+    public void testEverythingCalled() {
```
Comment (tkaitchuck) [inline shared/protocol/src/test/java/io/pravega/shared/protocol/netty/DelegatingRequestProcessorTest.java:29]: This doesn't actually assert anything. 
It should include an assertion that each method on the processor delegated to was invoked .

**Review by tkaitchuck — CHANGES_REQUESTED** [review]

**Review by andreipaduroiu — APPROVED** [review]

**Review by tkaitchuck — APPROVED** [review]

**Review by tkaitchuck — COMMENTED** [review]
Looks good. You just need to bump the wire command version, so that later we detect if these commands are present if needed.

**Comment by andreipaduroiu** [comment]
@tkaitchuck We've already bumped the wire command version for the 0.10 release. We'll only bump it up again if we won't be able to merge this feature before the 0.10 release cutoff.

**Pull request merged**